source/math/FloatBox2D.ooc

```diff
@@ -80,9 +87,19 @@ FloatBox2D: cover {
 		This new(left, top, width, height)
 	}
 	contains: func (point: FloatPoint2D) -> Bool {
-		this left <= point x && point x < this right && this top <= point y && point y < this bottom
+		this left <= point x && point x <= this right && this top <= point y && point y <= this bottom
 	}
 	contains: func ~box (box: FloatBox2D) -> Bool { this intersection(box) == box }
+	contains: func ~FloatPoint2DVectorList (list: FloatPoint2DVectorList) -> VectorList<Bool> {
+		result := VectorList<Bool> new()
+		for (i in 0..list count) {
+			if (this contains(list[i]))
+				result add(true)
+			else
+				result add(false)
+		}
+		result
+	}
```

Comment:
So then you create a list of Booleans that you directly afterwards send into a function that extracts the "true" values from the "false" values? That seems like a lot of unnecessary work. I would suggest doing the work done in `FloatBox2D contains~FloatPoint2DVectorList` directly where you use it (which is in only one place, as far as I can tell, and it shouldn't need to be longer than 4 lines if you only add the "true" elements).